addons/subscription/models/subscription.py

```diff
@@ -0,0 +1,127 @@
+    interval_type = fields.Selection([('days', 'Days'), ('weeks', 'Weeks'), ('months', 'Months')], string='Interval Unit', default='months')
+    exec_init = fields.Integer(string='Number of Documents')
+    date_init = fields.Datetime(string='First Date', default=fields.Datetime.now)
+    state = fields.Selection([('draft', 'Draft'), ('running', 'Running'), ('done', 'Done')], string='Status', copy=False, default='draft')
+    doc_source = fields.Reference(string='Source Document', required=True, selection=_get_document_types, size=128, help="User can choose the source document on which he wants to create documents", default=False)
+    doc_lines = fields.One2many('subscription.subscription.history', 'subscription_id', string='Documents created', readonly=True)
+    cron_id = fields.Many2one('ir.cron', string='Cron Job', help="Scheduler which runs on subscription", states={'running':[('readonly',True)], 'done':[('readonly', True)]})
+    note = fields.Text(string='Notes', help="Description or Summary of Subscription")
+
+    @api.model
+    def _auto_end(self):
+        super(SubscriptionSubscription, self)._auto_end()
+        # drop the FK from subscription to ir.cron, as it would cause deadlocks
+        # during cron job execution. When model_copy() tries to write() on the subscription,
+        # it has to wait for an ExclusiveLock on the cron job record, but the latter
+        # is locked by the cron system for the duration of the job!
+        # FIXME: the subscription module should be reviewed to simplify the scheduling process
+        #        and to use a unique cron job for all subscriptions, so that it never needs to
+        #        be updated during its execution.
+        self.env.cr.execute("ALTER TABLE %s DROP CONSTRAINT %s" % (self._table, '%s_cron_id_fkey' % self._table))
+
+    @api.multi
+    def set_process(self):
+        for row in self:
+            mapping = {'name':'name', 'interval_number':'interval_number', 'interval_type':'interval_type', 'exec_init':'numbercall', 'date_init':'nextcall'}
+            res = {'model':'subscription.subscription', 'args': repr([[row['id']]]), 'function':'model_copy', 'priority':6, 'user_id':row['user_id']}
+            for key,value in mapping.items():
+                res[value] = row[key]
+            cron = self.env['ir.cron'].create(res)
+            row.write({'cron_id':cron.id, 'state':'running'})
+
+    @api.multi
+    def model_copy(self):
+        for row in self:
+            if not row.get('cron_id',False):
+                continue
+            try:
+                id = row['doc_source'].id
+                model_name = row['doc_source']._name
+            except:
+                raise UserError(_('Please provide another source document.\nThis one does not exist!'))
+
+            default = {'state':'draft'}
+            document = self.env['subscription.document'].search([('model.model', '=', model_name)],limit=1)
```

Comment:
I disagree, if only used one, we can keep it here as long as it doesn't reduce the readability (e.g. making the line too long)